src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java

```diff
@@ -188,56 +147,62 @@ void canUpdateProductTest() throws Exception {
                 .andExpect(jsonPath("$.archived").value(productDtoDummy.isArchived()))
                 .andExpect(jsonPath("$.price").value(1.00));
 
-        verify(productService, times(1)).save(productDtoDummy);
+        verify(productService, times(1)).save(any(Product.class));
     }
 
     @Test
     @WithMockUser
-    void retrieveAllProductsFoundBySearchRequestTest() throws Exception {
+    void retrieveAllCategoriesTest() throws Exception {
         // given
-        String searchRequest = "POst";
-        ProductList dummyProductList = productMapperUtility.convertToProductListObject(List.of(createNonArchivedProduct()));
-        when(productService.findAllByNameContainingIgnoreCaseAndArchivedFalse(searchRequest)).thenReturn(dummyProductList);
+        final List<Category> categories = createCategoryList();
+        final CategoryList categoryListDummy = categoryMapperUtility.convertToCategoryList(categories);
+        when(categoryService.findAll()).thenReturn(categoryListDummy);
 
-        // when
+        // then
         ResultActions resultActions =
                 mockMvc.perform(
-                        get("/products/search/" + searchRequest)
+                        get("/products/categories")
                                 .contentType(MediaType.APPLICATION_JSON)
-                                .content(jsonUtils.asJsonString(dummyProductList.getProducts())));
+                                .content(jsonUtils.asJsonString(categoryListDummy.getCategories())));
+
         // then
         resultActions
                 .andExpect(status().isOk())
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON))
-                .andExpect(jsonPath("$.products", hasSize(1)))
-                .andExpect(jsonPath("$.products[0].name").value(equalTo("post-its")))
-                .andExpect(jsonPath("$.products[0].price").value(equalTo(1.00)));
+                .andExpect(jsonPath("$.categories", hasSize(1)))
+                .andExpect(jsonPath("$.categories[0].name").value(equalTo("Headphones")));
 
-        verify(productService, times(1)).findAllByNameContainingIgnoreCaseAndArchivedFalse(searchRequest);
+        verify(categoryService, times(1)).findAll();
     }
 
     @Test
     @WithMockUser
-    void retrieveAllCategoriesTest() throws Exception {
+    void retrieveAllProductsBySearchModelTest() throws Exception {
         // given
-        List<Category> categories = createCategoryList();
-        CategoryList categoryListDummy = categoryMapperUtility.convertToCategoryList(categories);
-        when(categoryService.findAll()).thenReturn(categoryListDummy);
+        final SearchModelDto searchModelDtoDummy = getDummySearchModelDto();
+        final SearchModel searchModelDummy = getDummySearchModel();
```

Comment:
better test would be: setting stricter searchModel. But that's your area of expertise, so just pretend I didn't say anything 😄

Reply:
my thought was: its a unit test so the content doesn't matter for testing this method, its a dependency for this method and its state doesn't matter
for tests in the service it would make sense, or for a <insert name of other tests that i cannot remember now>